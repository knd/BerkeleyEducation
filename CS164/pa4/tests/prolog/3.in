miserable(them).
schadenfreude(X) :- miserable(Y), ~miserable(X).
schadenfreude(me)?