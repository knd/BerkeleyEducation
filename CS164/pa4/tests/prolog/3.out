[(0, 1, 'R -> [a-z][a-zA-Z0-9]*'),
 (0, 4, 'Goal -> R \\( Args \\)'),
 (0, 5, 'Rule -> Goal \\.'),
 (0, 5, 'Rules -> Rule'),
 (0, 21, 'Rules -> Rules Rule'),
 (0, 26, 'S -> Rules Query'),
 (2, 3, 'A -> Value'),
 (2, 3, 'Args -> A'),
 (2, 3, 'R -> [a-z][a-zA-Z0-9]*'),
 (2, 3, 'Value -> [a-z][a-zA-Z0-9]*'),
 (5, 6, 'R -> [a-z][a-zA-Z0-9]*'),
 (5, 9, 'Goal -> R \\( Args \\)'),
 (5, 21, 'Rule -> Goal \\:\\- SubGoals \\.'),
 (7, 8, 'A -> Var'),
 (7, 8, 'Args -> A'),
 (7, 8, 'Var -> [A-Z][a-zA-Z_0-9]*'),
 (10, 11, 'R -> [a-z][a-zA-Z0-9]*'),
 (10, 14, 'G -> Goal'),
 (10, 14, 'Goal -> R \\( Args \\)'),
 (10, 14, 'SubGoals -> G'),
 (10, 20, 'SubGoals -> SubGoals \\, G'),
 (12, 13, 'A -> Var'),
 (12, 13, 'Args -> A'),
 (12, 13, 'Var -> [A-Z][a-zA-Z_0-9]*'),
 (15, 20, 'G -> \\~ Goal'),
 (16, 17, 'R -> [a-z][a-zA-Z0-9]*'),
 (16, 20, 'Goal -> R \\( Args \\)'),
 (18, 19, 'A -> Var'),
 (18, 19, 'Args -> A'),
 (18, 19, 'Var -> [A-Z][a-zA-Z_0-9]*'),
 (21, 22, 'R -> [a-z][a-zA-Z0-9]*'),
 (21, 25, 'Goal -> R \\( Args \\)'),
 (21, 26, 'Query -> Goal \\?'),
 (23, 24, 'A -> Value'),
 (23, 24, 'Args -> A'),
 (23, 24, 'R -> [a-z][a-zA-Z0-9]*'),
 (23, 24, 'Value -> [a-z][a-zA-Z0-9]*')]
Yes
